LLK/scratch-www
# [Develop] Add nested categories to project report module

### Changes:

This adds the ability to nest report categories in the project reporting module according to this design:
![project-report-mock-final](https://user-images.githubusercontent.com/4612592/78256643-d0951800-74c6-11ea-898f-8338ade55d04.png)

As a first round of changes, we have added nesting to only the Other option. The intended copy is over here: https://docs.google.com/document/d/13w9MbBrzb4NiztA9MUjiVEP4xdgUA9jz0zzisLIsnhQ/edit

When you report a project, all report categories will take you to a text field except for Other, which will take you to another dropdown.

The Back button will come in a later PR .

### Testing:

We will test these changes with people in the Sustainable Moderation project using this test plan: 

- Go to any project you do not own and report it. All categories you select should work as before except “Other”

- If you select “Other,” you should be prompted to make another selection. Some of those selections should result in a deadend; others will prompt you to input text

- Check that all the copy makes sense, is formatted correctly, and matches the copy written by the team

- Log into an admin account. Check that reports (that do not deadend) create tickets and that the text in the tickets makes sense for the new selections. Reports that do deadend should not create tickets.


cc @paulkaplan
